app/controllers/concerns/application_shared.rb

```diff
@@ -15,4 +15,35 @@ def set_timezone
     # Reset timezone for the next thread
     Time.zone = default_timezone
   end
-end
\ No newline at end of file
+
+  private
+
+  # Get the allowed parameters from the Apipie documentation in the API controllers.
+  def safe_params
+    apipie_params = Apipie.app.get_method_description("v2##{controller_name}", action_name).to_json[:params]
```

Comment:
This should be more defensive if the method description can't be found, e.g. the controller or action name doesn't match, as it'll return nil.

I'm also concerned that it's hard to override without overriding the whole method - it should be split up a little, or use more getters to determine the API name, so plugins or unusual core cases can override it partially.